src/tutorials/0008-introduction-to-libp2p/05.md

```diff
@@ -0,0 +1,80 @@
+---
+    title: "Multiple environments supported"
+    type: "text"
+---
+
+`libp2p` has multiple implementations and they all interop with each other.
+At the time of writing this, libp2p has 7 native implementations:
+
+<div class="flex justify-center items-center flex-wrap mv2">
+    <a href="https://github.com/libp2p/js-libp2p" target="__blank" class="flex justify-center items-center no-underline ma2 navy ph4-ns ph3 pv3-ns pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L05-js-logo.png" class="mr2  w2">
+        <div class="f4 b">js-libp2p</div>
+    </a>
+    <a href="https://github.com/libp2p/go-libp2p" target="__blank" class="flex justify-center items-center no-underline ma2 navy ph4-ns ph3 pv3-ns pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L05-go-lang-logo.png" class="mr2  w2">
+        <div class="f4 b">go-libp2p</div>
+    </a>
+    <a href="https://github.com/libp2p/rust-libp2p" target="__blank" class="flex justify-center items-center no-underline ma2 navy ph4-ns ph3 pv3-ns pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L05-rust-logo.png" class="mr2  w2">
+        <div class="f4 b">rust-libp2p</div>
+    </a>
+    <a href="https://github.com/libp2p/py-libp2p" target="__blank" class="flex justify-center items-center no-underline ma2 navy ph4-ns ph3 pv3-ns pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L05-py-logo.png" class="mr2  w2">
+        <div class="f4 b">py-libp2p</div>
+    </a>
+    <a href="https://github.com/libp2p/jvm-libp2p" target="__blank" class="flex justify-center items-center no-underline ma2 navy ph4-ns ph3 pv3-ns pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L05-jvm-logo.png" class="mr2  w2">
+        <div class="f4 b">jvm-libp2p</div>
+    </a>
+    <a href="https://github.com/libp2p/cpp-libp2p" target="__blank" class="flex justify-center items-center no-underline ma2 navy ph4-ns ph3 pv3-ns pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L05-cpp-logo.png" class="mr2  w2">
+        <div class="f4 b">cpp-libp2p</div>
+    </a>
+    <a href="https://github.com/status-im/nim-libp2p" target="__blank" class="flex justify-center items-center no-underline ma2 navy ph4-ns ph3 pv3-ns pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L05-nim-logo.svg" class="mr2  w2">
+        <div class="f4 b">nim-libp2p</div>
+    </a>
+</div>
+
+
+
+## Browser support
+
+
+The javascript implementation works in browsers and mobile browsers too! This is very important because it enables applications to run `libp2p` in the desktop and mobile devices as well.
+
+<div class="flex justify-center items-center mv4">
+    <img class="w-60" src="/tutorial-assets/T0008L05-libp2p-browser.png">
+    <img class="w-30" src="/tutorial-assets/T0008L05-libp2p-browser-mobile.png">
+</div>
+
```

Comment:
One important thing I have in mind here are breaking changes. For example, in 5 we are talking about the configuration of libp2p. We intend to completely change this next year (not soon!) to offer a smoother experience to our users while configuring the underlying modules. Also, we will probably create a set of ready to go configurations taht users could just import. Only advanced users would tune in the configuration.

So, with this in mind, we should have a way in this course of stating what version of libp2p this is considering. We will be looking into updating this over time, but it will eventually happen that we ship libp2p releases and it will take some time for the changes to be propagated here.

Reply:
Sounds good.
We can add a note to the tutorial description about the version of libp2p we are using in the tutorial. When we update the tutorial we can then add a message to warn users about the change and which version is now being used 👍

Reply:
Done! c692936da6f1740df418a3de4b4171ca6c86d123